test/functional/shared/resources/gateway_test.go

```diff
@@ -463,17 +461,45 @@ func testGatewayAvailability(hostname, baseURL, path string, expectedStatusCode
 	req.Host = hostname
 
 	client := newTestHTTPClient(isHttps, hostname)
-	res, err := client.Do(req)
-	if err != nil {
-		return err
-	}
 
-	if res.StatusCode != expectedStatusCode {
-		return fmt.Errorf("expected status code %d, got %d", expectedStatusCode, res.StatusCode)
+	retries := 2
+	retryBackoff := 5 * time.Second
+	for i := 1; i <= retries; i++ {
+		res, err := client.Do(req)
+		if err == nil && res.StatusCode == expectedStatusCode {
+			// Got expected status code, return
+			return nil
+		}
+
+		// If we got an error, or the status code was not what we expected, log the error and retry
+		// Logging the request and response will help with debugging the issue
+
+		t.Logf("failed to make request to %s with error: %s", urlPath, err)
+		requestDump, err := httputil.DumpRequestOut(req, true)
```

Comment:
this library looks useful but I didn't see it used anywhere else. anyone have any other ideas for if we should/ how we should log the request and response?

Reply:
we are using httputil library in ucp, for reverse proxy related constructions.  https://github.com/project-radius/radius/blob/b5310bc433529747e3bba810373e47b7df78670f/pkg/ucp/proxy/types.go

So perhaps this is good.